src/test/java/hudson/plugins/sshslaves/agents/AgentConnectionBase.java

```diff
@@ -46,11 +46,15 @@ public class AgentConnectionBase {
   @ClassRule
   public static CheckIsDockerAvailable isDockerAvailable = new CheckIsDockerAvailable();
 
-  @Rule
+  @Rule(order = 10)
   public JenkinsRule j = new JenkinsRule();
 
-  @Rule
-  public Timeout globalTimeout= new Timeout(4, TimeUnit.MINUTES);
+  @Rule(order = -10)
```

Comment:
Default order is -1 so this seems to make it apply to `@Rule GenericContainer` in subclasses.